src/screens/SendToken/SendTokenConfirm.js

```diff
@@ -19,108 +19,96 @@
 */
 import * as React from 'react';
 import { Keyboard } from 'react-native';
-import type { NavigationScreenProp } from 'react-navigation';
-import { connect } from 'react-redux';
 import t from 'translations/translate';
+import { useNavigation, useNavigationParam } from 'react-navigation-hooks';
 
 // constants
 import { SEND_TOKEN_PIN_CONFIRM } from 'constants/navigationConstants';
+import { CHAIN } from 'constants/chainConstants';
 
 // components
 import Table, { TableRow, TableLabel, TableAmount, TableTotal, TableUser, TableFee } from 'components/Table';
 import ContainerWithHeader from 'components/Layout/ContainerWithHeader';
 import Button from 'components/Button';
 import { Spacing, ScrollWrapper } from 'components/Layout';
 import TokenReviewSummary from 'components/ReviewSummary/TokenReviewSummary';
+import { BaseText } from 'components/Typography';
+
+// utils
+import { useChainsConfig } from 'utils/uiConfig';
 
 // types
-import type { RootReducerState } from 'reducers/rootReducer';
+import { useRootSelector } from 'selectors';
 
 
-type Props = {
-  navigation: NavigationScreenProp<*>,
-  session: Object,
-};
+const SendTokenConfirm = () => {
+  const session = useRootSelector(({ session: sessionState }) => sessionState.data);
+  const navigation = useNavigation();
 
-class SendTokenConfirm extends React.Component<Props> {
-  source: string;
+  const source = useNavigationParam('source');
+  const transactionPayload = useNavigationParam('transactionPayload');
```

Comment:
Pls add explicit typing to these params, consider adding some fallback code if they come null/undefined

```suggestion
  const source = useNavigationParam('source');
  const transactionPayload = useNavigationParam('transactionPayload');
```